dandi/support/digests.py

```diff
@@ -137,3 +137,42 @@ def md5file_nocache(filepath: str | Path) -> str:
     present in Zarrs
     """
     return Digester(["md5"])(filepath)["md5"]
+
+
+DIGEST_PRIORITY = ("dandi-etag", "zarr-checksum", "sha512", "sha256", "sha1", "md5")
+"""
+Default order to check file digests against.
+"""
+
+
+def check_digests(
+    filepath: str | Path,
+    digests: dict[str, str],
+    priority: tuple[str, ...] = DIGEST_PRIORITY,
+) -> bool:
+    """
+    Given some set of digests from DANDI metadata, eg that given by
+    :meth:`~dandi.download.Downloader.download_generator` to ``_download_file`` ,
+    check that a file matches.
+
+    Only check the first digest that is present in ``digests`` , rather than
+    checking all and returning if any are ``True`` - save compute time and
+    honor the notion of priority - some hash algos are better than others.
```

Comment:
```suggestion
    Only the first digest algorithm in ``priority`` that is present in
    ``digests`` is checked.
```

Describing an approach we're not doing is confusing, and "why" reasons generally don't belong in docstrings.

Reply:
I usually find "why" explanations helpful - eg. why not have a function that just take one hash rather than a dict of hashes, and if we are passing multiple hashes why wouldn't we check all of them - but sure i'll move that out of the docstring, no strong opinions. the test should protect against future memory loss

Reply:
updated docstring to also describe params and that makes previous description of what we're not doing obsolete anyway